src/Application/Command/CreateMilestones.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\AutomaticReleases\Application\Command;
+
+use InvalidArgumentException;
+use Laminas\AutomaticReleases\Git\Value\SemVerVersion;
+use Laminas\AutomaticReleases\Github\Api\GraphQL\Query\GetGithubMilestone;
+use Laminas\AutomaticReleases\Github\Api\V3\CreateMilestone;
+use Laminas\AutomaticReleases\Github\Event\Factory\LoadCurrentGithubEvent;
+use Laminas\AutomaticReleases\Github\Value\RepositoryName;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class CreateMilestones extends Command
+{
+    private LoadCurrentGithubEvent $loadEvent;
+    private GetGithubMilestone $getMilestone;
+    private CreateMilestone $createMilestone;
+
+    public function __construct(
+        LoadCurrentGithubEvent $loadEvent,
+        GetGithubMilestone $getMilestone,
+        CreateMilestone $createMilestone
+    ) {
+        parent::__construct('laminas:automatic-releases:create-milestones');
+
+        $this->loadEvent       = $loadEvent;
+        $this->getMilestone    = $getMilestone;
+        $this->createMilestone = $createMilestone;
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $milestoneClosedEvent = ($this->loadEvent)();
+        $repositoryName       = $milestoneClosedEvent->repository();
+        $releaseVersion       = $milestoneClosedEvent->version();
+        $milestone            = ($this->getMilestone)($repositoryName, $milestoneClosedEvent->milestoneNumber());
+
+        $milestone->assertAllIssuesAreClosed();
+
+        $this->createMilestoneIfNotExists($repositoryName, $releaseVersion->nextPatch());
+        $this->createMilestoneIfNotExists($repositoryName, $releaseVersion->nextMinor());
+        $this->createMilestoneIfNotExists($repositoryName, $releaseVersion->nextMajor());
+
+        return 0;
+    }
+
+    private function createMilestoneIfNotExists(RepositoryName $repositoryName, SemVerVersion $version): bool
+    {
+        try {
+            ($this->createMilestone)($repositoryName, $version->nextPatch());
+        } catch (InvalidArgumentException $e) {
+            // Log that the milestone exists?
```

Comment:
I have searched the GitHub API docs and haven't found a way to search for a milestone by its title. I've tried manually to create milestones and they failed for duplicate titles. I'm hoping that the API works the same. This needs to be tested.

Reply:
I can vouch for the API failing when the milestone already exists (I ran into this with keep-a-changelog).

Reply:
> I have searched the GitHub API docs and haven't found a way to search for a milestone by its title.

I've even looked for graphql queries to do this, but haven't found any. Every single example I've seen has you list milestones, and then do an array reduce to find the one matching the title you want.

Reply:
I'm not sure if loading all milestones and go through them is a good solution if you have a huge amount of milestones. So yes, if we are sure the API fails, we can go with that.

Reply:
100% positive it results in an error, so that's the best and easiest path.

Reply:
I'm doing it differently. GitHub sends a 201 status code if successful. Otherwise I'll throw an exception and pass through the response body for debugging.